src/main/java/Tree/TernaryTree.java

```diff
@@ -0,0 +1,161 @@
+package Tree;
+
+import Base.Tree;
+import Node.TernaryNode;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+
+public class TernaryTree<T> extends Tree {
+    TernaryNode<T> root;
+    TernaryNode<T> current;
+
+    public TernaryTree() { root = current = null; }
+
+    public TernaryTree(int key, T value) {
+        current = new TernaryNode<T>(key,value);
+        root = current;
+    }
+
+    protected void moveToLeftNode() {
+        assert current == null : "current must not be null";
+        current = current.getLeftNode();
+    }
+
+    protected void moveToRightNode() {
+        assert current == null : "current must not be null";
+        current = current.getRightNode();
+    }
+
+    protected void moveToMiddleNode() {
+        assert current == null : "current must not be null";
+        current = current.getRightNode();
+    }
+
+    protected void moveToParentNode() {
+        assert current == null : "current must not be null";
+        current = current.getParentNode();
+    }
+
+    protected void linkNodes(TernaryNode<T> parent, TernaryNode<T> child, char lmr) {
+        assert parent == null : "Parent node must not be null";
+        assert (lmr != 'R' || lmr != 'L' || lmr != 'M') : "Input must be R or L or M";
+        if (lmr == 'L') {
+            parent.setLeftNode(child);
+            if (child != null) child.setParentNode(parent);
+        } else if (lmr == 'R') { // lr =='R'
+            parent.setRightNod(child);
+            if (child != null) child.setParentNode(parent);
+        } else {
+            parent.setMiddletNode(child);
+            if (child != null) child.setParentNode(parent);
+        }
+    }
+
+    protected TernaryNode<T> getNode(int key) {
```

Comment:
This is invalid operation for TernaryTree.
Checking keys to move left or right is only valid for BinarySearchTree